### PR TITLE
Add API action to refresh external governance status.

### PR DESCRIPTION
**Pull request by guptanitish-ms**
## ARM API Information (Control Plane)

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow.
</i>
Azure 1st Party Service can try out the [Shift Left](https://aka.ms/ShiftLeft) experience to initiate API design review from ADO code repo.  If you are interested, may request engineering support by filling in with the form https://aka.ms/ShiftLeftSupportForm.

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [ ] new API version
      - [] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [x] Other, please clarify
      Add action to 2022-08-01 (Version yet to be released)
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month. 
Asap, we will be GA soon, currently in public preview.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  Asap, we will be GA soon, currently in public preview.
  4. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

### Contribution checklist (MS Employees Only):
- [x] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[x] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [x] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/sql/resource-manager/Microsoft.Sql/preview/2022-08-01-preview/examples/RefreshExternalGovernanceStatus.json
Comment (raosuhas) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2022-08-01-preview/examples/RefreshExternalGovernanceStatus.json:17]: I dont see this response defined in your api call

File: specification/sql/resource-manager/Microsoft.Sql/preview/2022-08-01-preview/examples/RefreshExternalGovernanceStatus.json
Comment (raosuhas) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2022-08-01-preview/examples/RefreshExternalGovernanceStatus.json:20]: You should not be reutrning full objects as the response for a POST call , it is better to respond with some details on the status of the operation

Comment (guptanitish-ms) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2022-08-01-preview/examples/RefreshExternalGovernanceStatus.json:20]: Which details do you think I should omit? Currently, the response does contain the status of the operation.

The return type is similar to:
https://learn.microsoft.com/en-us/rest/api/sql/2021-02-01-preview/servers/import-database?tabs=HTTP

File: specification/sql/resource-manager/Microsoft.Sql/preview/2022-08-01-preview/examples/RefreshExternalGovernanceStatus.json
Comment (guptanitish-ms) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2022-08-01-preview/examples/RefreshExternalGovernanceStatus.json:17]: I dont fully understand. Whats your microsoft alias so that I may ping you there.

Comment (guptanitish-ms) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2022-08-01-preview/examples/RefreshExternalGovernanceStatus.json:17]: Fixed this issue. The 200 response to reference the schema type.

Comment (raosuhas) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2022-08-01-preview/examples/RefreshExternalGovernanceStatus.json:17]: Resolved

File: specification/sql/resource-manager/Microsoft.Sql/preview/2022-08-01-preview/examples/RefreshExternalGovernanceStatus.json
Comment (raosuhas) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2022-08-01-preview/examples/RefreshExternalGovernanceStatus.json:20]: ok that makes sense. 
Although you need to fix the type here, the id indicates that this is a resource of type : 
"Microsoft.Sql/locations/refreshExternalGovernanceStatusOperationResults"

 but you have defined it as         "type": "Microsoft.Sql/servers/refreshExternalGovernanceStatusOperationResults"

Please fix either the Id or type as appropriate. The example you have shared has deined it correctly in the sample response.

Comment (guptanitish-ms) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2022-08-01-preview/examples/RefreshExternalGovernanceStatus.json]: Thanks, fixed.

**Comment by ericshape** [comment]
Can we ask for an exception for these failed rules? we will fix them in the next API version. 



❌ ResourceNameRestriction | The resource name parameter 'serverName' should be defined with a 'pattern' restriction.Location: Microsoft.Sql/preview/2022-08-01-preview/Servers.json#L413 | RPC-Uri-V1-05
-- | -- | --
❌ PostOperationAsyncResponseValidation | An async POST operation must set long running operation options 'x-ms-long-running-operation-options'Location: Microsoft.Sql/preview/2022-08-01-preview/Servers.json#L414 | RPC-Async-V1-11

**Review by ericshape — APPROVED** [review]

**Comment by guptanitish-ms** [comment]
> Can we ask for an exception for these failed rules? we will fix them in the next API version.
> 
> ❌ ResourceNameRestriction	The resource name parameter 'serverName' should be defined with a 'pattern' restriction.Location: Microsoft.Sql/preview/2022-08-01-preview/Servers.json#L413	RPC-Uri-V1-05
> ❌ PostOperationAsyncResponseValidation	An async POST operation must set long running operation options 'x-ms-long-running-operation-options'Location: Microsoft.Sql/preview/2022-08-01-preview/Servers.json#L414	RPC-Async-V1-11

Need exception for ResourceNameRestriction and LROLocation header (both errors already exist in the Servers.json in the past and are approved exceptions per ericshape).

**Comment by mikekistler** [comment]
Marking as "Approved-BreakingChange" only to unblock merge into the feature branch. But the breaking change to the x-ms-enum.name of the "publicNetworkAccess" property must be fixed before this can be merged to an official branch.

**Comment by ericshape** [comment]
I am OK with merging it.

**Comment by raych1** [comment]
> > Can we ask for an exception for these failed rules? we will fix them in the next API version.
> > ❌ ResourceNameRestriction	The resource name parameter 'serverName' should be defined with a 'pattern' restriction.Location: Microsoft.Sql/preview/2022-08-01-preview/Servers.json#L413	RPC-Uri-V1-05
> > ❌ PostOperationAsyncResponseValidation	An async POST operation must set long running operation options 'x-ms-long-running-operation-options'Location: Microsoft.Sql/preview/2022-08-01-preview/Servers.json#L414	RPC-Async-V1-11
> 
> Need exception for ResourceNameRestriction and LROLocation header (both errors already exist in the Servers.json in the past and are approved exceptions per ericshape).

approved lint in this PR.

**Pull request merged**
